doc/sphinx/language/extensions/compil-steps.rst

```diff
@@ -0,0 +1,103 @@
+.. _compil-steps:
+
+=============================
+How Coq Processes Input Terms
+=============================
+
+It is in theory possible to write down every Gallina term explicitly
+as described in the :ref:`Core Language <core-language>` part of this
+manual, for instance
+:n:`Nat.add (S O) (Nat.mul (S (S O)) (S (S O)))`. However, this would
+be very tedious and error-prone and takes us away from our usual
+mathematical practice. To circumvent this, Rocq offers multiple
+preprocessing mechanisms to help fill the gap between what the users
+would like to input to the system and the fully formal core language
+expected by the kernel. We give an overview of all these steps below.
+
+For instance, the notation mechanisms reflects the eponymous mathematical
+practice and allows to write :n:`1 + 2 * 2` instead of the above
+term. Those mechanisms range from simple :ref:`Abbreviations` to full
+fledged :ref:`Notations` with user defined :ref:`syntaxes
+<ReservingNotations>`. Multiple interpretations can be given to the
+same syntax in different contexts thanks to the :ref:`scope
+<Scopes>` mechanism. For instance :n:`(1 + 2 * 2)%N` can be
+the above natural number expression while :n:`(1 + 2 * 2)%Z` can be
+an expression with integers.
+
+In order to take the best part of all these preprocessing mechanisms,
+one needs a basic understanding of the multiple steps needed to
+transform an inputed term (possibly with notations) into the valid
+Gallina term which Rocq will ultimately use internally. Terms given as input
+to Rocq go through several successive steps:
+
+* During :gdef:`lexing` the input is split into a sequence of tokens,
+  for instance the string ``"1 + 2 * 2"`` is split into the
+  sequence of five tokens `'1' '+' '2' '*' '2'`. A set of :ref:`basic
+  tokens <lexical-conventions>` are predefined and can be extended, in
+  particular by :ref:`reserving notations <ReservingNotations>`.
```

Comment:
The second sentence jumps to a much greater level of detail, especially the part about reserving notations.  It's a distraction to include it here.  You might want to add that detail to the `lexical-conventions` section.  And referring to the lexical-conventions section is fine.

Reply:
Well, it's both important to know that there are some hardcoded tokens and that this set is extensible. We could maybe spare the link to how the extension can be done, but not sure this would spare much.